sqrl-engines/sqrl-engine-server/src/main/java/com/datasqrl/engine/server/GenericJavaServerEngine.java

```diff
@@ -73,11 +88,64 @@ public EnginePhysicalPlan plan(StagePlan plan, List<StageSink> inputs, Execution
     Preconditions.checkArgument(dbStages.size()==1, "Currently only support a single database stage in server");
     ExecutionEngine engine = Iterables.getOnlyElement(dbStages).getEngine();
     Preconditions.checkArgument(engine instanceof JDBCEngine, "Currently the server only supports JDBC databases");
-    return new ServerPhysicalPlan(((ServerStagePlan) plan).getModel(), ((JDBCEngine)engine).getConnector());
+
+    ServerConfig updatedConfig = applyDefaults(serverConfig, ((JDBCEngine)engine).getConnector(), this.port);
+    return new ServerPhysicalPlan(((ServerStagePlan) plan).getModel(), updatedConfig);
   }
 
-  @Override
-  public EnginePhysicalPlan readPlanFrom(Path directory, String stageName, Deserializer deserializer) throws IOException {
-    return ServerPhysicalPlan.readFrom(directory, stageName, deserializer);
+  public static ServerConfig applyDefaults(ServerConfig serverConfig, JdbcDataSystemConnector connector, int port) {
+    if (connector.getDialect().equals("postgres")) {
+      if (serverConfig.getPgConnectOptions() == null) {
+        serverConfig.setPgConnectOptions(new PgConnectOptions());
+      }
+      PgConnectOptions pgConnect = serverConfig.getPgConnectOptions();
+
+      if (connector.getHost() != null) {
+        pgConnect.setHost(connector.getHost());
+      }
+      if (connector.getPort() != null) {
+        pgConnect.setPort(connector.getPort());
+      }
+      if (connector.getUser() != null) {
+        pgConnect.setUser(connector.getUser());
+      }
+      if (connector.getPassword() != null) {
+        pgConnect.setPassword(connector.getPassword());
+      }
+      if (connector.getDatabase() != null) {
+        pgConnect.setDatabase(connector.getDatabase());
+      }
+    }
```

Comment:
IMHO it's a good idea to add an else clause in these cases to make it easier to spot the error once we add additional dialects. A simple `else { throw NotYetImplementedException(); }` is good enough so it doesn't fail silently.